ui/pages/asset/components/asset-chart.tsx

```diff
@@ -0,0 +1,289 @@
+      const index = Math.max(
+        0,
+        Math.min(
+          data.length - 1,
+          Math.round((chart.crosshairX / chart.width) * data.length),
+        ),
+      );
+
+      const point = data[index];
+      if (point) {
+        const { x: xAxis, y: yAxis } = chart.scales;
+        const x = xAxis.getPixelForValue(point.x);
+        const y = yAxis.getPixelForValue(point.y);
+
+        chart.ctx.lineWidth = 1;
+        chart.ctx.strokeStyle = '#BBC0C5';
+        chart.ctx.beginPath();
+        chart.ctx.moveTo(x, 0);
+        chart.ctx.lineTo(x, chart.height);
+        chart.ctx.stroke();
+
+        chart.ctx.beginPath();
+        chart.ctx.arc(x, y, 3, 0, 2 * Math.PI);
+        chart.ctx.fillStyle = chart.options.borderColor as string;
+        chart.ctx.fill();
+      }
+    }
+  },
+};
+
+Chart.register(
+  LinearScale,
+  PointElement,
+  LineElement,
+  Filler,
+  Decimation,
+  crosshairPlugin,
+);
+
+const initialChartOptions: ChartOptions<'line'> & { fill: boolean } = {
+  normalized: true,
+  parsing: false,
+  aspectRatio: 2.4,
+  layout: { autoPadding: false, padding: 0 },
+  animation: { duration: 0 },
+  fill: true,
+  backgroundColor: ({ chart }) => {
+    const gradient = chart.ctx.createLinearGradient(0, 0, 0, chart.height);
+    gradient.addColorStop(0, `${chart.options.borderColor}60`);
+    gradient.addColorStop(1, `${chart.options.borderColor}00`);
+    return gradient;
+  },
+  elements: {
+    line: { borderWidth: 2 },
+    point: { pointStyle: false },
+  },
+  plugins: {
+    decimation: {
+      algorithm: 'lttb',
+      samples: 150,
+      threshold: 150,
+      enabled: true,
+    },
+  },
+};
+
+// A chart showing historic prices for a native or token asset
+const AssetChart = ({
+  chainId,
+  address,
+  currentPrice,
+  currency,
+}: {
+  chainId: `0x${string}`;
+  address: string;
+  currentPrice?: number;
+  currency: string;
+}) => {
+  const t = useI18nContext();
+  const theme = useTheme();
+  const [timeRange, setTimeRange] = useState<TimeRange>('1D');
+
+  const chartRef = useRef<Chart<'line', Point[]>>();
+  const priceRef = useRef<{
+    setPrice: (_: { price?: number; date?: number }) => void;
+  }>();
+
+  const {
+    loading,
+    data: { prices, edges },
+  } = useHistoricalPrices({
+    chainId,
+    address,
+    currency,
+    timeRange,
+  });
+
+  const { xMin, xMax, yMin, yMax } = edges ?? {};
+  const options = {
+    ...initialChartOptions,
+    borderColor: theme === 'dark' ? brandColor.blue400 : brandColor.blue500,
+    scales: {
+      x: { min: xMin?.x, max: xMax?.x, display: false, type: 'linear' },
+      y: { min: yMin?.y, max: yMax?.y, display: false },
+    },
+  } as const;
+
+  const noData = !loading && !prices;
+
+  return (
+    <Box
+      className={classnames({
+        'asset__chart--no-data': noData && !currentPrice,
+        'asset__chart--no-data-dark':
+          noData && !currentPrice && theme === 'dark',
```

Comment:
any way to select dark theme in css instead?  Need the placeholder gradient to depend on theme.